cmake/Modules/FindSYCLToolkit.cmake

```diff
@@ -49,12 +49,27 @@ find_file(
   )
 
 # Find SYCL library fullname.
-find_library(
-  SYCL_LIBRARY
-  NAMES sycl
-  HINTS ${SYCL_LIBRARY_DIR}
-  NO_DEFAULT_PATH
-)
+# On both Linux and Windows, the current version of SYCL runtime is 7.
+# On Linux, libsycl.so symbolic links to libsycl.so.7.
+# On Windows, sycl7.dll version suffix is used.
+foreach(sycl_lib_version "" 7)
```

Comment:
Pls submit a JIRA to track the requirement to oneAPI.

Reply:
Thanks, will submit. Future reviewer, see https://github.com/pytorch/pytorch/pull/126725#discussion_r1616909975 for background.

Reply:
If it is Windows specific limitation, I would suggest we keep the code for Linux unchanged. Add comments on why Windows is special and what we plan to do in the future.

Reply:
Thanks, updated. It is Windows specific limitation.